packages/ses/test/test-scope.js

```diff
@@ -0,0 +1,341 @@
+      set(_value) {
+        globalObjectProtoSetterValue = this;
+      },
+    },
+    globalObjectProtoFn: {
+      value() {
+        return this;
+      },
+      configurable: true,
+      writable: true,
+    },
+    globalObjectProtoFnOptimizable: {
+      value() {
+        return this;
+      },
+      configurable: false,
+      writable: false,
+    },
+  });
+  const globalObject = Object.create(globalObjectProto, {
+    globalObjectGetter: {
+      get() {
+        return this;
+      },
+    },
+    globalObjectSetter: {
+      set(_value) {
+        globalObjectSetterValue = this;
+      },
+    },
+    globalObjectFn: {
+      value() {
+        return this;
+      },
+      configurable: true,
+      writable: true,
+    },
+    globalObjectFnOptimizable: {
+      value() {
+        return this;
+      },
+      configurable: false,
+      writable: false,
+    },
+  });
+  const globalLexicalsProto = Object.create(null, {
+    globalLexicalsProtoGetter: {
+      get() {
+        return this;
+      },
+    },
+    globalLexicalsProtoSetter: {
+      set(_value) {
+        globalLexicalsProtoSetterValue = this;
+      },
+    },
+    globalLexicalsProtoFn: {
+      value() {
+        return this;
+      },
+      configurable: true,
+      writable: true,
+    },
+    globalLexicalsProtoFnOptimizable: {
+      value() {
+        return this;
+      },
+      configurable: false,
+      writable: false,
+    },
+  });
+  const globalLexicals = Object.create(globalLexicalsProto, {
+    globalLexicalsGetter: {
+      get() {
+        return this;
+      },
+    },
+    globalLexicalsSetter: {
+      set(_value) {
+        globalLexicalsSetterValue = this;
+      },
+    },
+    globalLexicalsFn: {
+      value() {
+        return this;
+      },
+      configurable: true,
+      writable: true,
+    },
+    globalLexicalsFnOptimizable: {
+      value() {
+        return this;
+      },
+      configurable: false,
+      writable: false,
+    },
+  });
+
+  const knownScopeProxies = new WeakSet();
+  const { safeEvaluate: evaluate } = makeSafeEvaluator({
+    globalObject,
+    globalLexicals,
+    knownScopeProxies,
+  });
+
+  // Known compromise in fidelity of the emulated script environment (all tests):
+
+  t.is(evaluate('globalObjectProtoGetter'), globalObject);
+  t.is(evaluate('globalObjectGetter'), globalObject);
+  t.is(evaluate('globalLexicalsProtoGetter'), globalObject);
+  t.is(evaluate('globalLexicalsGetter'), globalObject);
+
+  evaluate('globalObjectProtoSetter = 123');
+  t.is(globalObjectProtoSetterValue, globalObject);
+  evaluate('globalObjectSetter = 123');
+  t.is(globalObjectSetterValue, globalObject);
+  // bug: properties in prototype of globalLexicals error on set
+  t.throws(() => evaluate('globalLexicalsProtoSetter = 123'), {
+    instanceOf: Error,
+  });
+  t.is(globalLexicalsProtoSetterValue, undefined);
+  evaluate('globalLexicalsSetter = 123');
+  t.is(globalLexicalsSetterValue, globalObject);
```

Comment:
I'm confused, I would have expected the scope proxy here, not the `globalObject`.

Is that because we use an explicit `Reflect.get` / `Reflect.set` with the `globalObject`?

Reply:
I was also surprised by the results from this test!

you are correct about the reason for `globalLexicalsSetterValue` https://github.com/endojs/endo/blob/c603c5aa4a1ba271cf17d754df789a52aa7debfb/packages/ses/src/scope-handler.js#L122-L124